### PR TITLE
Tree editor improvements

### PR DESCRIPTION
**Pull request by lucas-koehler**
* Reduce vertical spacing between controls by updating `theia-tree-editor` dependency.
* Add maximum width for detail view content. Center content in detail view.
* Extend example data to be free of validation errors

Note: Should be squashed on merge

### CODE REVIEW

File: templates/tree-editor/tree-frontend-module.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import 'theia-tree-editor/style/index.css';
 import 'theia-tree-editor/style/forms.css';
+// Custom style sheets should be imported after all default stylesheets
+// because for conflicting styles, last imported ones win.
```
Comment (sdirix) [inline templates/tree-editor/tree-frontend-module.ts]: Do we need this comment? Feels redundant.

File: templates/tree-editor/style/editor.css [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.theia-tree-editor-form {
+  display: flex;
+  justify-content: center;
+}
+
+.theia-tree-editor-form > .jsonforms-container {
+  flex-grow: 1;
+  max-width: 960px; /* Half the width of a full hd display. */
+}
```
Comment (sdirix) [inline templates/tree-editor/style/editor.css]: Would be nice to have a newline

**Review by sdirix — COMMENTED** [review]
In general it works fine :+1: As the form is now restricted in size it feels weird that there is so much space between the tree and the form if there is more space available. Should we move the form to the left instead of rendering it in center?

File: templates/tree-editor/tree-frontend-module.ts (hunk 1 above)
Comment (lucas-koehler) [reply templates/tree-editor/tree-frontend-module.ts]: True, this is just a general css principle and not specific to the editor. I'll remove it

**Review by sdirix — APPROVED** [review]
Great! Thanks!

**Pull request merged**